osu.Game.Rulesets.Osu.Tests/Editor/TestSceneOsuEditorGrids.cs

```diff
@@ -44,6 +44,28 @@ public void TestGridToggles()
             rectangularGridActive(false);
         }
 
+        [Test]
+        public void TestDistanceSnapMomentaryToggle()
+        {
+            AddStep("select second object", () => EditorBeatmap.SelectedHitObjects.Add(EditorBeatmap.HitObjects.ElementAt(1)));
+
+            AddUntilStep("distance snap grid hidden", () => !this.ChildrenOfType<OsuDistanceSnapGrid>().Any());
+            AddStep("hold alt", () => InputManager.PressKey(Key.AltLeft));
+            AddUntilStep("distance snap grid visible", () => this.ChildrenOfType<OsuDistanceSnapGrid>().Any());
+            AddStep("release alt", () => InputManager.ReleaseKey(Key.AltLeft));
+            AddUntilStep("distance snap grid hidden", () => !this.ChildrenOfType<OsuDistanceSnapGrid>().Any());
+        }
+
+        [Test]
+        public void TestGridSnapMomentaryToggle()
+        {
+            rectangularGridActive(false);
+            AddStep("hold shift", () => InputManager.PressKey(Key.ShiftLeft));
+            rectangularGridActive(true);
+            AddStep("release shift", () => InputManager.ReleaseKey(Key.ShiftLeft));
```

Comment:
Merging #20903 broke this test. `rectangularGridActive` is relying on `Key.Number2` to activate the placement tool:

https://github.com/ppy/osu/blob/181c7d1d6e9900225321f6170948bbe96767ac08/osu.Game.Rulesets.Osu.Tests/Editor/TestSceneOsuEditorGrids.cs#L49

But because in this case the step is surrounded by <kbd>Shift</kbd> presses, now this turns into a <kbd>Shift</kbd>-<kbd>2</kbd> which previously did nothing, but now changes the current snap and does not change the current tool to placement, making the rest of the test fail.

Reply:
I've fixed this in a.. maybe unorthodox way. But it works.